sdk/src/connections/usb/windows/usb_eeprom_windows.cpp

```diff
@@ -101,6 +103,42 @@ Status UsbEeprom::write(const uint32_t address, const uint8_t *data,
             << hr;
         return Status::GENERIC_ERROR;
     }
+
+    ExUnitHandle handle;
+
+    hr = UsbWindowsUtils::UvcFindNodeAndGetControl(
+        &handle, &m_implData->handle->pVideoInputFilter);
+    if (hr != S_OK) {
+        LOG(WARNING) << "Failed to find node and get control. Error: "
+                     << std::hex << hr;
+        return Status::GENERIC_ERROR;
+    }
+
+    uint8_t eepromWriteStatus = 0;
+    int attempts = 300;
+
+    while (eepromWriteStatus == 0 && attempts > 0) {
+        hr = UsbWindowsUtils::UvcExUnitGetProperty(&handle, 7,
+                                                   &eepromWriteStatus, 1);
+        --attempts;
+        if (FAILED(hr)) {
+            LOG(WARNING)
+                << "Failed to read a property via UVC extension unit. Error: "
+                << hr;
+            return Status::GENERIC_ERROR;
+        }
+
+        if (!eepromWriteStatus) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
The SDK might be used in an application with a lot of threads (parallel image processing). Doing a busy wait (sleep) here would just block the current thread. Maybe use `std::this_thread::yield()`, allowing other threads to run. https://en.cppreference.com/w/cpp/thread/yield

Reply:
I will give it a try with this. One concern is that if there are very few other threads, the time checking and yielding will load the CPU while the sleep will not use CPU.

Reply:
Also, writing the EEPROM is something that is usually done during the initialization of the board and not while capturing and processing frames. In this case, using yield() will not do any good.